benchmarks/definition/regression/core.py

```diff
@@ -0,0 +1,494 @@
+    correlation.
+
+    Args:
+        x: First array of values.
+        y: Second array of values, same shape as x.
+
+    Returns:
+        Kendall's Tau correlation coefficient.
+    """
+    tau, _ = kendalltau(x, y)
+    return tau
+
+
+def spearman_rho_score(x: np.ndarray, y: np.ndarray, /) -> float:
+    """Calculate Spearman's Rho correlation coefficient.
+
+    Values close to 1 indicate strong positive monotonic correlation,
+    values close to -1 indicate strong negative monotonic correlation.
+
+    Args:
+        x: First array of values.
+        y: Second array of values, same shape as x.
+
+    Returns:
+        Spearman's Rho correlation coefficient.
+    """
+    rho, _ = spearmanr(x, y)
+    return rho
+
+
+# Dictionary mapping transfer learning model names to their surrogate classes
+TL_MODELS = {
+    "index_kernel": GaussianProcessSurrogate,
+}
+
+
+# List of regression metrics to evaluate model performance
+REGRESSION_METRICS = [
+    root_mean_squared_error,
+    mean_squared_error,
+    r2_score,
+    mean_absolute_error,
+    max_error,
+    explained_variance_score,
+    kendall_tau_score,
+    spearman_rho_score,
+]
+
+
+def run_tl_regression_benchmark(
+    settings: TransferLearningRegressionBenchmarkSettings,
+    data_loader: DataLoader,
+    searchspace_factory: SearchSpaceFactory,
+    objective_factory: ObjectiveFactory,
+) -> pd.DataFrame:
+    """Run a transfer learning regression benchmark.
+
+    This function evaluates the performance of transfer learning models compared to
+    non transfer learning models on regression tasks. It varies the fraction of
+    source data used for training and the number of training points in the target task.
+
+    For each combination, it trains a surrogate on target data only and transfer
+    learning models on source + target data, then evaluates their predictive
+    performance on held-out target test data using the provided regression metrics.
+
+    Args:
+        settings: The benchmark settings.
+        data_loader: Function that loads the dataset for regression evaluation.
+        searchspace_factory: Function that creates search spaces for both
+            non-TL and TL model scenarios.
+        objective_factory: Function that creates the optimization objective.
+
+    Returns:
+        DataFrame with benchmark results containing performance metrics for each
+        model, training scenario, and Monte Carlo iteration. Columns include:
+
+      - scenario: Model scenario identifier (e.g., "0_reduced_searchspace",
+        "5_index_kernel")
+      - Performance metrics: root_mean_squared_error, mean_squared_error, r2_score,
+        mean_absolute_error, max_error, explained_variance_score, kendall_tau_score,
+        spearman_rho_score
+      - Experimental metadata: mc_iter, n_train_pts, fraction_source, n_source_pts,
+        n_test_pts, source_data_seed
+
+    """
+    objective = objective_factory()
+    data = data_loader()
```

Comment:
Perhaps this a stupid question that comes way too late but:

If the only purpose of these two factories is that they are called without any context (i.e. taking no arguments) right at the beginning of the function and then never used again --> why do you use factories in the first place!? 🤔 You could have just passed the objects directly, no?